pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -326,11 +326,31 @@ func (c *Cluster) postReconcileUpdateOSDProperties(desiredOSDs map[int]*OSDInfo)
 	}
 	logger.Debugf("post processing osd properties with %d actual osds from ceph osd df and %d existing osds found during reconcile", len(osdUsage.OSDNodes), len(desiredOSDs))
 	for _, actualOSD := range osdUsage.OSDNodes {
-		if desiredOSD, ok := desiredOSDs[actualOSD.ID]; ok {
-			if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
-				// Log the error and allow other updates to continue
-				logger.Error(err)
-			}
+		desiredOSD, ok := desiredOSDs[actualOSD.ID]
+		if !ok {
+			return nil
```

Comment:
we need to continue the loop
```suggestion
			continue
```